src/store/options.js

```diff
@@ -178,13 +179,15 @@ const Options = {
       // Sync if the current memory context get options for the first time
       if (!prevOptions) sync(options);
 
-      observers.forEach(async (fn) => {
-        try {
-          await fn(options, prevOptions);
-        } catch (e) {
-          console.error(`[options] Error while observing options: `, e);
+      resolvedObservers = (async () => {
```

Comment:
this looks like it may be super random. if there are multiple sources of option mutation a customer have to be lucky to call `resolveObservers` at the right time

Reply:
No, it's not. The background process has only one instance of options, and this method is called only once, after the options are updated.